src/hooks/useMaxSiblingWidth.ts

```diff
@@ -0,0 +1,68 @@
+import { useCallback, useEffect, useMemo, useState } from 'react'
+import Path from '../@types/Path'
+import editingValueStore from '../stores/editingValue'
+import parentOf from '../util/parentOf'
+
+/**
+ * Helper function to collect nodes with the specified `data-parent-path`.
+ * If no thought node is found, it continues searching with `data-grand-parent-path`.
+ */
+const collectNodes = (path: string): HTMLElement[] => {
+  /** Collect nodes by provided attribute and value. */
+  const collectByAttribute = (attribute: string, value: string): HTMLElement[] => {
+    return Array.from(document.querySelectorAll<HTMLElement>(`[data-${attribute}='${value}']`)).filter(
+      node => !node.querySelector('.divider'),
+    )
+  }
+
+  let collectedNodes = collectByAttribute('parent-path', path)
+
+  // If no nodes found, fallback to data-grand-parent-path
+  if (!collectedNodes.length) {
```

Comment:
The only child status is already available in React, so we could avoid the extra DOM query.

Reply:
Not sure if I understood this correctly. Don't really have a way to figure out if it is the only child due to the structure of all thoughts being flat. Let me know if I am missing something here.

Reply:
The thoughts are flat only after they have been linearized for rendering. If you go up the component hierarchy to [LayoutTree](https://github.com/cybersemics/em/blob/5ffafd3a22beb062ff2b55b87115fe898aea7870/src/components/LayoutTree.tsx), you have access to hierarchical data in `linearizeTree`. You can make the check there, and then pass it down as a prop to the Divider.

All components also have access to hierarchical thought data through various Redux selectors. It's possible in the Divider to create a selector that uses [findAnyChild](https://github.com/cybersemics/em/blob/5ffafd3a22beb062ff2b55b87115fe898aea7870/src/selectors/getChildren.ts#L172) on `parentOf(path)` to see if any non-divider sibling exists. Let me know if you need additional explanation.